presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -0,0 +1,424 @@
+                    requiredJoinSymbols.stream().filter(leftSymbols::contains).collect(toImmutableList()));
+            if (leftResult.cost.hasUnknownComponents()) {
+                return UNKNOWN_COST_RESULT;
+            }
+            if (leftResult.cost.equals(INFINITE_COST)) {
+                return INFINITE_COST_RESULT;
+            }
+            PlanNode left = leftResult.planNode.orElseThrow(() -> new IllegalStateException("no planNode present"));
+            JoinEnumerationResult rightResult = getJoinSource(
+                    ImmutableList.copyOf(rightSources),
+                    requiredJoinSymbols.stream()
+                            .filter(rightSymbols::contains)
+                            .collect(toImmutableList()));
+            if (rightResult.cost.hasUnknownComponents()) {
+                return UNKNOWN_COST_RESULT;
+            }
+            if (rightResult.cost.equals(INFINITE_COST)) {
+                return INFINITE_COST_RESULT;
+            }
+            PlanNode right = rightResult.planNode.orElseThrow(() -> new IllegalStateException("no planNode present"));
+
+            // sort output symbols so that the left input symbols are first
+            List<Symbol> sortedOutputSymbols = Stream.concat(left.getOutputSymbols().stream(), right.getOutputSymbols().stream())
+                    .filter(outputSymbols::contains)
+                    .collect(toImmutableList());
+
+            // Cross joins can't filter symbols as part of the join
+            // If we're doing a cross join, use all output symbols from the inputs and add a project node
+            // on top
+            List<Symbol> joinOutputSymbols = sortedOutputSymbols;
+            if (joinConditions.isEmpty() && joinFilters.isEmpty()) {
+                joinOutputSymbols = Stream.concat(left.getOutputSymbols().stream(), right.getOutputSymbols().stream())
+                        .collect(toImmutableList());
+            }
+
+            JoinEnumerationResult result = setJoinNodeProperties(new JoinNode(
+                    idAllocator.getNextId(),
+                    INNER,
+                    left,
+                    right,
+                    joinConditions,
+                    joinOutputSymbols,
+                    joinFilters.isEmpty() ? Optional.empty() : Optional.of(and(joinFilters)),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty()));
+
+            if (!joinOutputSymbols.equals(sortedOutputSymbols)) {
+                PlanNode resultNode = new ProjectNode(idAllocator.getNextId(), result.planNode.get(), identity(sortedOutputSymbols));
+                result = new JoinEnumerationResult(costProvider.getCumulativeCost(resultNode), Optional.of(resultNode));
+            }
+
+            return result;
+        }
+
+        private JoinEnumerationResult getJoinSource(List<PlanNode> nodes, List<Symbol> outputSymbols)
+        {
+            PlanNode planNode;
```

Comment:
`planNode` variable is not used outside of the `if` block; perhaps, move it inside of the `if` block